### PR TITLE
feat: pre commit linting

### LINKED ISSUE #65: Set up pre-commit linting
As discussed, having a pre-commit hook to run the linter for us would be really handy.

### PR DESCRIPTION
**Pull request by adamalton**
Allows us (developers) to discover our mistakes before we make them public!

Fixes #65.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,12 +10,20 @@
...
+    "husky": "^8.0.1",
+    "lint-staged": "^13.0.3"
   },
   "engines": {
     "node": "16.x",
     "npm": ">=7.x"
+  },
+  "lint-staged": {
+    "packages/api/**": ".husky/lint-api.sh",
+    "packages/client/**": "husky/lint-client.sh"
```
Comment (joshghent) [inline package.json:27]: Is this missing a `.` for the husky folder name?

**Review by joshghent — COMMENTED** [review]
Looks good to me, other than the file name

File: package.json (hunk 1 above)
Comment (adamalton) [reply package.json:27]: Ooh, maaaybe 😐

Comment (adamalton) [reply package.json]: Fixed.

**Pull request merged**